packages/components/src/presentation-components/table/UsePresentationTable.ts

```diff
@@ -84,10 +96,59 @@ export function usePresentationTable<TColumn, TRow>(props: UsePresentationTableP
  */
 export function usePresentationTableWithUnifiedSelection<TColumn, TRow>(
   props: Omit<UsePresentationTableProps<TColumn, TRow>, "keys">,
-): UsePresentationTableResult<TColumn, TRow> {
+): UsePresentationTableWithUnifiedSelectionResult<TColumn, TRow> {
   const unifiedSelection = useUnifiedSelectionContext();
   const keys = unifiedSelection?.getSelection() ?? emptyKeySet;
-  return usePresentationTable({ ...props, keys });
-}
+  const [selectedRows, setSelectedRows] = useState<TableRowDefinition[]>();
```

Comment:
This list of rows should always be in sync with unified selection at `+1` but it gets out of sync when `onSelect` is called. It should be updated when `onSelect` is replacing selection or `selectionChange` event handler should not skip selection changes caused by this hook.

Reply:
Fixed this by removing checking between tableGuid and event source, so when onSelect() is called, selectedRows get updated afterwards, therefore it does not go out of sync.